torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -3793,6 +3826,75 @@ std::unique_ptr<Function> CompilationUnit::define(
   return fn;
 }
 
+std::vector<Function*> CompilationUnit::define(
+    const c10::optional<c10::QualifiedName>& prefix,
+    const std::vector<Property>& properties,
+    const std::vector<ResolverPtr>& propResolvers,
+    const std::vector<Def>& definitions,
+    const std::vector<ResolverPtr>& defResolvers,
+    const Self* self,
+    bool shouldMangle) {
+  TORCH_INTERNAL_ASSERT(self);
+  TORCH_INTERNAL_ASSERT(definitions.size() == defResolvers.size());
+  TORCH_INTERNAL_ASSERT(properties.size() == propResolvers.size());
+  std::vector<Function*> functions;
+  std::unordered_map<std::string, Function*> function_table;
+
+  for (size_t i = 0; i < properties.size(); i++) {
+    std::pair<std::unique_ptr<Function>, std::unique_ptr<Function>>
+        property_fns = define(
+            prefix,
+            properties[i],
+            propResolvers[i],
+            self,
+            function_table,
+            shouldMangle);
+
+    auto& getter_fn = property_fns.first;
+    auto& setter_fn = property_fns.second;
+
+    const auto& getter_name = getter_fn->name();
```

Comment:
This four-line block of code gets repeated three times in this function, can we factor it out ?

Reply:
Factored it out into a local lambda.